src/NuGet.Clients/NuGet.CommandLine/Commands/ProjectFactory.cs

```diff
@@ -675,9 +675,19 @@ private PackageDependency CreateDependencyFromProject(dynamic project)
                     projectFactory.ProcessNuspec(builder, null);
                 }
 
+                VersionRange versionRange = VersionRange.Parse(builder.Version.ToString());
```

Comment:
I don't know if perf is critical here, but you don't have to create a string for the version and parse it if it's going to be overridden by nuspecVersion.